### PR TITLE
lightning: maintain task meta in singleTaskMetaMgr

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/34213

Problem Summary:

Lightning precheck doesn't check local disk resources and cluster available.

### What is changed and how it works?

When `incremental-import` is set to false, Lightning doesn't create a metadata schema in downstream TiDB. Instead, it creates a `singleTaskMetaMgr` to maintain task metadata. However, `singleTaskMetaMgr` always returns true for task existence check.

https://github.com/pingcap/tidb/blob/9fc73ac219cbf65151fb9e8a41699c7b1d6cadef/br/pkg/lightning/restore/meta_manager.go#L1088

Precheck for local disk resources and cluster available are accidentally skipped. Because lightning thinks the check has been performed before. 
https://github.com/pingcap/tidb/blob/9fc73ac219cbf65151fb9e8a41699c7b1d6cadef/br/pkg/lightning/restore/restore.go#L1923

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the issue that Lightning precheck doesn't check local disk resources and cluster available.
```

### CODE REVIEW

File: br/pkg/lightning/restore/meta_manager.go [hunk 1]
```diff
@@ -1068,15 +1071,32 @@ func (b singleMgrBuilder) TableMetaMgr(tr *TableRestore) tableMetaMgr {
 }
 
 type singleTaskMetaMgr struct {
-	pd *pdutil.PdController
+	pd           *pdutil.PdController
+	taskID       int64
+	initialized  bool
+	sourceBytes  uint64
+	clusterAvail uint64
```
Comment (glorv) [inline br/pkg/lightning/restore/meta_manager.go:1078]: I don't see any read from this field

Comment (gozssky) [reply br/pkg/lightning/restore/meta_manager.go:1078]: It may be read and updated by `CheckTasksExclusively`.  Although there is no place to actually read it, adding it will avoid bugs caused by future changes.

File: br/pkg/lightning/restore/meta_manager.go [hunk 2]
```diff
@@ -1068,15 +1071,32 @@ func (b singleMgrBuilder) TableMetaMgr(tr *TableRestore) tableMetaMgr {
...
 	return nil
 }
 
 func (m *singleTaskMetaMgr) CheckTasksExclusively(ctx context.Context, action func(tasks []taskMeta) ([]taskMeta, error)) error {
-	_, err := action(nil)
+	newTasks, err := action([]taskMeta{
+		{
+			taskID:      m.taskID,
+			status:      taskMetaStatusInitial,
+			sourceBytes: m.sourceBytes,
```
Comment (glorv) [inline br/pkg/lightning/restore/meta_manager.go:1092]: I think you should also set clusterAvail here to taskMeta

Comment (gozssky) [reply br/pkg/lightning/restore/meta_manager.go]: Done.

**Review by glorv — APPROVED** [review]

File: br/pkg/lightning/restore/meta_manager.go [hunk 3]
```diff
@@ -1068,15 +1071,34 @@ func (b singleMgrBuilder) TableMetaMgr(tr *TableRestore) tableMetaMgr {
...
 
 func (m *singleTaskMetaMgr) InitTask(ctx context.Context, source int64) error {
+	m.sourceBytes = uint64(source)
+	m.initialized = true
 	return nil
 }
 
 func (m *singleTaskMetaMgr) CheckTasksExclusively(ctx context.Context, action func(tasks []taskMeta) ([]taskMeta, error)) error {
-	_, err := action(nil)
+	newTasks, err := action([]taskMeta{
```
Comment (D3Hunter) [inline br/pkg/lightning/restore/meta_manager.go:1088]: seems changes in this file is not related to the fixing.

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: br/pkg/lightning/restore/meta_manager.go (hunk 3 above)
Comment (gozssky) [reply br/pkg/lightning/restore/meta_manager.go:1088]: When checking cluster resource, `sourceBytes` need to read by `CheckTasksExclusively`.
https://github.com/pingcap/tidb/blob/b7f26143e500379fec94c286badcac26bdfc90d6/br/pkg/lightning/restore/check_info.go#L127

**Review by buchuitoudegou — APPROVED** [review]
lgtm

**Review by D3Hunter — APPROVED** [review]

**Comment by gozssky** [comment]
/hold

Wait for the integration-br-test to pass.

**Comment by gozssky** [comment]
/unhold

All tests are passed.
https://ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/11342/pipeline/1612/
